src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -10,502 +11,314 @@
-                    C = new(null, 42, null, 3.14e+17f, null, true, null, new int[] { 42, 17 })
-                    {
-                        A = "Hello",
-                        J = new(null)
-                        {
-                            B = 7,
-                        },
-                    },
-                };
-                await JsonSerializer.SerializeAsync(stream, obj, new JsonSerializerOptions { Converters = { new OuterConverter<TestClass<NestedClass>>() } });
-            }
+                var reader = new Utf8JsonReader(sequence, true, default);
+                JsonSerializer.Deserialize(ref reader, type, readOptions);
+            });
+        }
 
-            stream.Position = 0;
+        private class Chunk : ReadOnlySequenceSegment<byte>
+        {
+            public Chunk(string json, int firstSegmentLength)
             {
-                var readOptions = new JsonSerializerOptions
+                Memory<byte> bytes = Encoding.UTF8.GetBytes(json);
+                Memory = bytes.Slice(0, firstSegmentLength);
+                RunningIndex = 0;
+                Next = new Chunk()
                 {
-                    DefaultBufferSize = 128,
-                    IgnoreNullValues = ignoreNullValues,
+                    Memory = bytes.Slice(firstSegmentLength),
+                    RunningIndex = firstSegmentLength,
+                    Next = null,
                 };
-
-                Outer<TestClassWithParamCtor<NestedClassWithParamCtor>> obj = await JsonSerializer.DeserializeAsync<Outer<TestClassWithParamCtor<NestedClassWithParamCtor>>>(stream, readOptions);
-
-                Assert.Equal(new string('x', paddingLength), obj.S);
-                Assert.Equal("Hello", obj.C.A);
-                Assert.Null(obj.C.B);
-                Assert.Equal(42, obj.C.C);
-                Assert.Null(obj.C.D);
-                Assert.Equal(3.14e17f, obj.C.E);
-                Assert.Null(obj.C.F);
-                Assert.True(obj.C.G);
-                Assert.Null(obj.C.H);
-                Assert.Collection(obj.C.I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));
-                Assert.Null(obj.C.J.A);
-                Assert.Equal(7, obj.C.J.B);
             }
+            private Chunk()
+            { }
         }
 
-        private static IEnumerable<object[]> TestData()
+        private interface ITestObject
         {
-            foreach (int length in ContinuationPaddingLengths)
-            {
-                foreach (bool ignore in IgnoreNullValues)
-                {
-                    yield return new object[] { length, ignore };
-                }
-            }
+            string A { get; set; }
+            string B { get; set; }
+            int C { get; set; }
+            int? D { get; set; }
+            float E { get; set; }
+            bool G { get; set; }
+            int[] I { get; set; }
+            INestedObject J { get; set; }
```

Comment:
Can you also add a dictionary of items property, maybe dictionary of a couple of `INestedObject`?